### PR TITLE
Import process: add/fix production names where needed

### LINKED ISSUE #117: Glyph names in sources - automatic renaming process
Let's create an automatic renaming process to make the glyph names from the FB sources consistent with https://github.com/googlefonts/googlesans-flex/blob/main/.github/actions/import/glyph-list.txt

Font Bakery fail: [Glyph names are all valid?](https://github.com/googlefonts/googlesans-flex/actions/runs/3707745278/jobs/6284525053#step:9:284)
@belluzj @madig 
CC @chrissimpkins

### PR DESCRIPTION
**Pull request by RickyDaMa**
Will close #117

If there is an existing postscript name, checks it doesn't have hyphens and removes them if so

If there isn't an existing postscript name, checks the design name doesn't have hyphens and adds a postscript name with none if so

### CODE REVIEW

**Comment by RickyDaMa** [comment]
Thought: should I just include this in the fix metadata script? I guess postscript names are metadata given they're a lib key

File: .github/actions/import/scripts/fix_glyph_production_names.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+ROOT_DIR = Path(__file__).parent.parent
+POSTSCRIPT_NAMES = "public.postscriptNames"
+
+
+def fix_production_names(designspace_path: Path):
+    designspace = DesignSpaceDocument.fromfile(designspace_path)
+    ufos: list[Font] = designspace.loadSourceFonts(Font.open)
+    for ufo in ufos:
+        production_names: Optional[dict[str, str]] = ufo.lib.get(POSTSCRIPT_NAMES)
```
Comment (madig) [inline .github/actions/import/scripts/fix_glyph_production_names.py]: Best to take prodnames from `designspace.findDefault().font`.

File: .github/actions/import/scripts/fix_glyph_production_names.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        for glyph in ufo:
+            if not glyph.name:
+                continue
+
+            has_production_name = production_names is not None and glyph.name in production_names
+            if has_production_name and "-" in production_names[glyph.name]:
+                existing_production_name = production_names[glyph.name]
```
Comment (madig) [inline .github/actions/import/scripts/fix_glyph_production_names.py]:
```suggestion
            has_production_name = production_names is not None and (existing_production_name := production_names.get(glyph.name))
            if has_production_name and "-" in existing_production_name:
```

File: .github/actions/import/scripts/fix_glyph_production_names.py (hunk 1 above)
Comment (RickyDaMa) [reply .github/actions/import/scripts/fix_glyph_production_names.py]: So production names only usually come from one UFO? Makes sense I guess, saves duplicating

**Comment by madig** [comment]
Yeah, including it in the fixer seems appropriate, especially because it's supposed to be a bandaid for getting 1.0 out the door.

File: .github/actions/import/scripts/fix_glyph_production_names.py (hunk 2 above)
Comment (RickyDaMa) [reply .github/actions/import/scripts/fix_glyph_production_names.py]: Done. Ugly, but done

File: .github/actions/import/scripts/fix_glyph_production_names.py (hunk 1 above)
Comment (RickyDaMa) [reply .github/actions/import/scripts/fix_glyph_production_names.py]: Done

File: .github/actions/import/scripts/fix_glyph_production_names.py [hunk 3]
```diff
@@ -26,15 +26,18 @@
...
             if not glyph.name:
                 continue
 
-            has_production_name = production_names is not None and glyph.name in production_names
-            if has_production_name and "-" in production_names[glyph.name]:
-                existing_production_name = production_names[glyph.name]
+            has_production_name = production_names is not None and (
+                existing_production_name := production_names.get(glyph.name)
+            )
+            if has_production_name and existing_production_name:
```
Comment (madig) [inline .github/actions/import/scripts/fix_glyph_production_names.py]: Oh shi I dropped the `"-" in` 😬 sorry

Comment (RickyDaMa) [reply .github/actions/import/scripts/fix_glyph_production_names.py]: I can't believe you've done this.

**Pull request merged**